### ISSUE TITLE
Sidebar close button

### ISSUE THREAD

**Issue by M374LX** [LMMS/lmms#3682]
A close button for the sidebar would be instantly recognizable, unlike the need to click on the currently selected option to hide the bar:
![2017-07-03-194642_1920x1080_scrot](https://user-images.githubusercontent.com/6584238/27809888-5b04d34a-602a-11e7-9863-8798ca7aaccc.png)

Another small benefit this would bring is that the mouse cursor, when located on any of the editors (as in most of the time), would have a shorter distance to travel.

**Comment by Hussam-Eddin-Alhomsi** [comment]
I don't like the current way of closing the open tab for this reason:
when the user opens a tab and performs a drag & drop action for a file, he will always do the action on the right side of the GUI, and to close the open tab, he will have to return all the way to the left.

But if we are going to change how tabs are closed, I think we should change how they open.

**Comment by tresf** [comment]
@Hussam-Eddin-Alhomsi, you'll have to be more specific.  The mouse dragging across the screen and back again paradox a common UI annoyance.  I'm not sure how that is specific to LMMS.

To avoid the sidebar is impossible given our current UI.  You should either explicitly stated how this can be improved within scope of this PR, refute this PR, or open and crosslink a new enhancement.  We embrace great ideas but they shouldn't get in the way of good ones.

**Comment by budislav** [comment]
Just add one more button in the main toolbar.
Is it possible to move tabs on the top, just icons?
![untitled](https://user-images.githubusercontent.com/3619927/28028915-7c07cd6c-659e-11e7-827f-46c464c2a931.png)

**Comment by Hussam-Eddin-Alhomsi** [comment]
> A couple of reasons why I propose this positioning of the buttons is that vertical reading is impossible so text there is useless. We have only icons and with only icons we have a lot of empty space. Space is precious so we can just show and hide browser with button on the top just like every other window.

I totally agree. 💯 
I like your mockup because it removes the icons from the far left and also removes the vertical text, and becuase you positioned the text/image of the open tab in a proper position, becuase it's currently positioned like this:
![image](https://user-images.githubusercontent.com/25503477/28088609-e105e5d2-668e-11e7-81d0-6c98b70db12d.png)

**Comment by budislav** [comment]
Yes and there is no need for two same icons, also I would remove title underline. 
Also that message "Drag an instrument into.." is unnecessary, because user need to read that only once, only first time he start LMMS, he will knew that every other time. But even if he didn't read that he can assume. That is not good place for short tutorial. It should be in note editor when user open LMMS for the first time.

**Comment by RebeccaLaVie** [comment]
@Umcaruje Curious to hear your thoughts on positioning the sidebar tabs at the top?

I would have to agree that rotated text does not lend to much of an improvement with readability when the tooltips already work well for users that are not sure what each icon represents yet.

**Comment by gi0e5b06** [comment]
I would put the sidebar on the right side. Much more practical, no flickering (give the option!).
Check Inkscape for a quite good sidebar ui.
I agree the vertical text is not readable and annoying, easy to remove. I would put the buttons in the main tool bar but on the right side (above the sidebar if it was on the right side). The close button is not useful if the selected tab is easily reconizable (i use a red border or a red background).

PS: We could remove the sidebar completely and use a subwindow instead. Technically not a problem.
![sidebar_20170724](https://user-images.githubusercontent.com/30031949/28544773-bc669542-70c4-11e7-97d9-baa09f05285a.jpg)

**Comment by Hussam-Eddin-Alhomsi** [comment]
This is a duplicate of #3234.

**Comment by BaraMGB** [comment]
Perhaps we close https://github.com/LMMS/lmms/issues/3234.

**Comment by Sawuare** [comment]
@gi0e5b06 

> I would put the sidebar on the right side. Much more practical, no flickering (give the option!).

The LMMS GUI is from left to right, I don't think positioning the sidebar on the right only is a good idea, but giving the option is probably a good idea (read below).

> PS: We could remove the sidebar completely and use a subwindow instead.

Is a subwindow the same as a normal window like the piano roll?
If it is, some benefits to the subwindow over the sidebar are movability and resizability from all directions, which is good, and that will also let you choose to have the browser on the left or the right side, or anywhere.

**Comment by gi0e5b06** [comment]
@Sawuare
1) The trend is to put such a sidebar on the right side. In the old days, it was more often on the left side. In all cases, it should be an option.
2) Yes. Also make things easier in the case of a tabbed ui.

**Comment by tresf** [comment]
>  The trend is to put such a sidebar on the right side. In the old days, it was more often on the left side. In all cases, it should be an option.

I use sidebars in dozens of applications.  They're overwhelmingly on the right.  This makes sense, UI is generally left to right, so navigation panes stick to the left.  There are a few exceptions to this when software puts the toolbar on the left instead, but left is an overwhelmingly adopted standard. 

Inkscape and Gimp both  use floating toolbars on the left, which is more common for graphical applications but not necessarily desktop applications in general.

**Comment by tresf** [comment]
Typo... 

> overwhelmingly on the **left**

**Comment by gi0e5b06** [comment]
Having use both, the sidebar on the right is a lot more confortable and practical.
Anyway, we could argue for years but that's just 5-7 lines of code to add the option. So let's just implement it and move on. And by the way, the best technical solution is the one used by Qt Designer (I think the sidebars are floatable toolbars you can move around).

**Comment by codevenkat** [comment]
Hello. I am a beginner. Do you think I can take up this issue?

**Comment by codevenkat** [comment]
If no one is working on it I will try to implement it.

**Comment by PhysSong** [comment]
Implemented via #5133.
